armi/reactor/converters/axialExpansionChanger/axialExpansionChanger.py

```diff
@@ -148,6 +83,71 @@ def __init__(self, detailedAxialExpansion: bool = False):
         self.linked = None
         self.expansionData = None
 
+    @classmethod
+    def expandColdDimsToHot(
+        cls,
+        assems: list,
+        isDetailedAxialExpansion: bool,
+        referenceAssembly=None,
+    ):
+        """Expand BOL assemblies, resolve disjoint axial mesh (if needed), and update block BOL heights.
+
+        .. impl:: Perform expansion during core construction based on block heights at a specified temperature.
+            :id: I_ARMI_INP_COLD_HEIGHT
+            :implements: R_ARMI_INP_COLD_HEIGHT
+
+            This method is designed to be used during core construction to axially thermally expand the
+            assemblies to their "hot" temperatures (as determined by ``Thot`` values in blueprints).
+            First, The Assembly is prepared for axial expansion via ``setAssembly``. In
+            ``applyColdHeightMassIncrease``, the number densities on each Component is adjusted to
+            reflect that Assembly inputs are at cold (i.e., ``Tinput``) temperatures. To expand to
+            the requested hot temperatures, thermal expansion factors are then computed in
+            ``computeThermalExpansionFactors``. Finally, the Assembly is axially thermally expanded in
+            ``axiallyExpandAssembly``.
+
+            If the setting ``detailedAxialExpansion`` is ``False``, then each Assembly gets its Block mesh
+            set to match that of the "reference" Assembly (see ``getDefaultReferenceAssem`` and ``setBlockMesh``).
+
+            Once the Assemblies are axially expanded, the Block BOL heights are updated. To account for the change in
+            Block volume from axial expansion, ``completeInitialLoading`` is called to update any volume-dependent
+            Block information.
+
+        Parameters
+        ----------
+        assems: list[:py:class:`Assembly <armi.reactor.assemblies.Assembly>`]
+            list of assemblies to be thermally expanded
+        isDetailedAxialExpansion: bool
+            If False, assemblies will be forced to conform to the reference mesh after expansion
+        referenceAssembly: :py:class:`Assembly <armi.reactor.assemblies.Assembly>`, optional
+            Assembly whose mesh other meshes will conform to if isDetailedAxialExpansion is False.
+            If not provided, will assume the finest mesh assembly which is typically fuel.
+
+        Notes
+        -----
+        Calling this method will result in an increase in mass via applyColdHeightMassIncrease!
+
+        See Also
+        --------
+        :py:meth:`armi.reactor.converters.axialExpansionChanger.axialExpansionChanger.AxialExpansionChanger.applyColdHeightMassIncrease`
+        """
+        assems = list(assems)
+        if not referenceAssembly:
+            referenceAssembly = getDefaultReferenceAssem(assems)
+        axialExpChanger = cls(isDetailedAxialExpansion)
```

Comment:
Why is this line necessary?

Is there some internal code that will be making use of this?

Reply:
I don't think I follow the question. I can try to explain the intent and maybe that will help.

The axial expansion specific behavior that existed in blueprints construction was a call to the `expandColdDimsToHot` function. Inside that function, it created an `AxialExpansionChanger` instance to perform the initial expansion of assemblies. The drawback is there is no way to alter _what_ performs the axial expansion inside that `expandColdDimsToHot` let alone modify the function call there.

By making this a classmethod on the base axial expander class and allowing plugins to provide their own expander subclass **if they wish**, we can provide more control to plugins and applications about how the expansion on blueprints is performed. If at all.

I could do something like
```python
class SillyAxialExpander(AxialExpansionChanger):
    def axiallyExpandAssembly(self):
        # do my own expansion on an assembly
```
or
```python
class NoOpExpander(AxialExpansionChanger):
    @classmethod
    def expandColdDimsToHot(cls, ...):
        # actually don't do anything
        return
```
Now if my plugin implements this hook and returns one of these classes, I have exercised my control over how to perform the initial expansion of assemblies. And potentially not at all in the case of the more explicit `NoOpExpander`

Note: Changes in https://github.com/terrapower/armi/commit/86be86c7c01f272ff78d02f334e22f18a6b59f40 mean allow the case where no plugin implements this hook to also not do any expansion.

Reply:
Okay, sure, I get that.

Thanks!